cpp/src/structure/create_graph_from_edgelist_impl.cuh

```diff
@@ -274,16 +274,12 @@ create_graph_from_edgelist_impl(
                   "Invalid input arguments: edgelist_srcs.size() != edgelist_dsts.size().");
   CUGRAPH_EXPECTS(!edgelist_weights || (edgelist_srcs.size() == (*edgelist_weights).size()),
                   "Invalid input arguments: edgelist_srcs.size() != edgelist_weights.size().");
-  CUGRAPH_EXPECTS(!edgelist_id_type_pairs ||
-                    (edgelist_srcs.size() == std::get<0>((*edgelist_id_type_pairs)).size()),
-                  "Invalid input arguments: edgelist_srcs.size() != "
-                  "std::get<0>((*edgelist_id_type_pairs)).size().");
-  CUGRAPH_EXPECTS(!edgelist_id_type_pairs ||
-                    (edgelist_srcs.size() == std::get<1>((*edgelist_id_type_pairs)).size()),
-                  "Invalid input arguments: edgelist_srcs.size() != "
-                  "std::get<1>((*edgelist_id_type_pairs)).size().");
   CUGRAPH_EXPECTS(renumber,
                   "Invalid input arguments: renumber should be true if multi_gpu is true.");
+  // edgelist_id_pairs may be mismatched depending on how edges were distributed.
+  // This function is responsible for redistributing the tuple.
+  // The sizes of that tuple are checked at the end of this function to ensure it was
+  // redistributed correctly.
```

Comment:
What is this? This function is assuming that edge lists (including edge types & IDs) are pre-shuffled.

Reply:
As @seunghwak mentioned, the C++ function here assumes the shuffling has been done before this call.

It appears to me (first glance at code I haven't looked at in 3 weeks) that the C API needs to be updated to shuffle this data properly.

The code that follows should fail if the edge type/ids are not shuffled with the edge src/dst pairs.  Moving the `CUGRAPH_EXPECTS` calls will mask this problem, not correct it.

Reply:
I would suggest reverting all changes in this file.  I can look at fixing the shuffling in #2997.